### PR TITLE
Backward compatibility & Design considerations sections

### PR DESCRIPTION
**Pull request by iritkatriel**

### CODE REVIEW

**Comment by iritkatriel** [comment]
This PR has two commits - a small one to edit "backward compatibility" section. The paragraph I removed is now copied to https://github.com/python/exceptiongroups/issues/19, pending decision about whether to include asyncio integation in this PEP.

The second commit is larger - it rewrites the "design considerations" section (why we did X instead of Y)  as "rejected ideas" (why did we reject Y and do X instead).

**Review by gvanrossum — APPROVED** [review]
I do wonder whether we need to say more about backwards compatibility. E.g. if someone has code that uses `except Exception:` and some library starts wrapping its exceptions in ExceptionGroup, that won't be caught. While that's really the library that's being backwards incompatible, I think we should call that out.

**Comment by iritkatriel** [comment]
> I do wonder whether we need to say more about backwards compatibility. E.g. if someone has code that uses `except Exception:` and some library starts wrapping its exceptions in ExceptionGroup, that won't be caught. While that's really the library that's being backwards incompatible, I think we should call that out.

Good point, let me put that bit back.

File: except_star.md [hunk 1]
```diff
@@ -835,6 +836,22 @@ Programs will only be impacted by the changes proposed in this PEP once they
...
+are derived from `BaseException`. Any such clause will need to be replaced
+by `except (Exception, ExceptionGroup):` or `except *Exception:`.
+
+* Similarly, any `except T:` clause that wraps code which is now potentially
+raising `ExceptionGroup` needs to become `except *T:`, and its body may need
+to be updated.
+
+* Libraries that need to support older python versions will not be able to use
+`except*`, and may need to handle `ExceptionGroup`s raised in user code through
+the low-low_level `ExceptionGroup` APIs.
```
Comment (gvanrossum) [inline except_star.md:852]: I think it's stronger. They may not *raise* `ExceptionGroup`.

Comment (iritkatriel) [reply except_star.md:852]: I think what @1st1  meant with this was that you can:

```
try:
    userCode()
except BaseException as e:
    if python_version >= 3.11:
           if it's an exception group: do stuff to it
    do whatever else
```

Comment (iritkatriel) [reply except_star.md:852]: Otherwise what is there to say? "if you need to support older python versions then you can't use the new features" ?

Comment (gvanrossum) [reply except_star.md]: Sorry, I didn't notice this bullet is specifically about older Python versions.

And when I think about the case I was concerned about more, that is just about calling out that raising ExceptionGroup from a library API may break applications that use that API and should thus be considered a new feature of the library. So I think we're good.

**Review by gvanrossum — APPROVED** [review]

**Pull request merged**